File: src/main/java/com/plugin/utils/RandomAccessFileUtil.java
package com.plugin.utils;

import static com.plugin.utils.SystemUtil.println;

import java.io.RandomAccessFile;

import kotlin.text.Charsets;

public class RandomAccessFileUtil {

    public void change(String path, String mode) {
        try (RandomAccessFile raf = new RandomAccessFile(path, mode)) {
            long point = 0;
            while (raf.getFilePointer() != raf.length()) {

                String reads = new String(raf.readLine().getBytes(Charsets.ISO_8859_1), Charsets.UTF_8);
                println("reads: " + reads);

                // 插入数据
                insert(raf, reads, point);
                point = raf.getFilePointer();
            }
        } catch (Exception e) {
            println("读取文件异常: " + e.getMessage());
        }
    }

    public void insert(RandomAccessFile raf, String content, long index) {
        if (content.contains("\"")) {
            try {
                raf.seek(index);
                String change = "魔改诗句";
                String result = change + content;
//                raf.write(result.getBytes(Charsets.UTF_8));
                raf.writeUTF(result);
            } catch (Exception e) {
                println("插入数据异常：" + e.getMessage());
            }
        }
    }


}

File: RandomAccessFileUtil.kt
package utils

import java.io.RandomAccessFile
import java.nio.charset.StandardCharsets

class RandomAccessFileUtil {
    fun readFile(path: String?, mode: String?): ArrayList<String> {
        val list = ArrayList<String>()
        try {
            RandomAccessFile(path, mode).use { raf ->
                // RandomAccessFile默认使用的是【iso-8859-1】字符集，使用的时候，需要把他转换成UTF-8
                var currentPointer: Long = 0
                val length = raf.length()
                while (currentPointer < length) {
                    val filePointer = raf.filePointer
                    val line = raf.readLine()
                    if (line != null) {
                        val reads = String(line.toByteArray(StandardCharsets.ISO_8859_1), StandardCharsets.UTF_8)
                        list.add(reads)
                        // println("read:" + reads + " filePointer:" + currentPointer);
                    }
                    currentPointer = filePointer
                }
                raf.close()
            }
        } catch (e: Exception) {
            println("read file failed: " + e.message)
        }
        return list
    }
}